### PR TITLE
Move projects to different directories

### PR DESCRIPTION
**Pull request by jaredpar**
The SDK / NuGet does not support projects existing in the same directory during restore. It creates non-deterministic behavior. I noticed odd behavior while tracking down a test failure in #71686. Digging into the output it appeared our build was mixing outputs and steppping back I noticed the double projects in the same directory.

This change updates BuildBoss to catch this mistake and refactors our project layout to avoid the problem.

https://github.com/dotnet/sdk/issues/28763

### CODE REVIEW

**Comment by jaredpar** [comment]
@tmat going to need you to look over the interactive host changes. Those were the hardest to factor out.

**Comment by jaredpar** [comment]
@dotnet/roslyn-infrastructure PTAL

**Comment by sharwell** [comment]
> The SDK / NuGet does not support projects existing in the same directory during restore.

💭 This seems strange. Under default configurations, there are definitely conflicts, but you can instruct the build to avoid it by using some special properties in the project.

File: src/Interactive/HostProcess/x86/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Explicitly empty global editorconfig file to ensure we aren't accidentally applying unexpected rules to interactive documents
+global = true
```
Comment (sharwell) [inline src/Interactive/HostProcess/x86/.editorconfig:2]: I think you meant one of these:

```suggestion
root = true
```

```suggestion
is_global = true
```

**Comment by sharwell** [comment]
Based on the way Arcade configures output paths (BaseIntermediateOutputPath), I'm not sure the premise here is correct. Our build will confuse files if two projects in different directories have the same name (e.g. src/A/Project.csproj and src/B/Project.csproj), but it shouldn't confuse files if projects in the same directory have different names.

**Comment by jaredpar** [comment]
> I'm not sure the premise here is correct. 

It's not my premise: it's the stated behavior by the SDK and NuGet team. 

This particular bug cost me and R9 architect over a day of build frustration and investigation. That is what lead to me filing the SDK bug and finding it was unsupported. Earlier this week I was investigating a spurious infra failure where our build output was non-sensical and it was coming from one of our places where we have multiple project files in the same directory. Given the race style issue this can create it's not something I want to mess around with.

I _deeply wish_ that SDK / MSBuild / NuGet issued a warning here because it is such an easy pattern to fall into and has such hard to track down consequences. But that will likely have to wait until .NET 9 / msbuild analyzers.

File: src/CodeStyle/Core/Tests/Microsoft.CodeAnalysis.CodeStyle.LegacyTestFramework.UnitTestUtilities.csproj [hunk 2]
```diff
@@ -1,40 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (tmat) [inline src/CodeStyle/Core/Tests/Microsoft.CodeAnalysis.CodeStyle.LegacyTestFramework.UnitTestUtilities.csproj:1]: Odd that git doesn't see this as a rename.

**Review by tmat — APPROVED** [review]

File: src/CodeStyle/Core/Tests/Microsoft.CodeAnalysis.CodeStyle.LegacyTestFramework.UnitTestUtilities.csproj (hunk 2 above)
Comment (jaredpar) [reply src/CodeStyle/Core/Tests/Microsoft.CodeAnalysis.CodeStyle.LegacyTestFramework.UnitTestUtilities.csproj:1]: I was surprised as well.

File: src/Interactive/HostProcess/x86/.editorconfig (hunk 1 above)
Comment (jaredpar) [reply src/Interactive/HostProcess/x86/.editorconfig:2]: This file was just moved, not created.

**Pull request merged**